In [8]:
from math import factorial

In [9]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [10]:
def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

In [11]:
def permutations(n):
    return int(factorial(n))

**Задача 1**

Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, если в этот день должно быть 4 разных занятия?

In [13]:
combinations(10, 4)

210

Ответ: 210

**Задача 2**

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

In [21]:
one_ace = combinations(4, 1)
one_ace_rest = combinations(48, 3)
one_ace_combinations = one_ace * one_ace_rest
one_ace_combinations

69184

In [25]:
two_ace = combinations(4, 2)
two_ace_rest = combinations(48, 2)
two_ace_combinations = two_ace * two_ace_rest
two_ace_combinations

6768

In [26]:
three_ace = combinations(4, 3)
three_ace_rest = combinations(48, 1)
three_ace_combinations = three_ace * three_ace_rest
three_ace_combinations

192

In [27]:
four_ace = combinations(4, 4)
four_ace_rest = combinations(48, 0)
four_ace_combinations = four_ace * four_ace_rest
four_ace_combinations

1

In [28]:
answer = one_ace_combinations + two_ace_combinations + three_ace_combinations + four_ace_combinations
answer

76145

Ответ: 76145

**Задача 3**

Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что два определённых человека будут сидеть рядом?

In [37]:
combinations(7, 2)

21

Ответ: 21

**Задача 4**

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

In [35]:
Pa = combinations(50, 3) * combinations(10, 0) / combinations(52, 3)
Pa

0.8868778280542986

In [36]:
Pb = combinations(50, 2) * combinations(10, 1) / combinations(52, 3)
Pb

0.5542986425339367

Ответ: а)0.8868778280542986, б) 0.5542986425339367

**Задача 5**

Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

События независимые, т.к. вероятность любого из них не зависит от появления, либо непоявления другого события.

**Задача 6**

Допустим, имеется некоторая очень редкая болезнь (поражает 0.1 % населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99 % больных этой болезнью и всего лишь в 1 % случаев даёт ложный положительный ответ.

Вопрос: какова вероятность, что вы действительно больны ей?

Вероятность заболевания - 0,001
Вероятность незаболевания - 0,999
Вероятность неверного теста - 0,01
Вероятность верного теста - 0,99


In [41]:
PaB = (0.999 * 0.01) / (0.01 * 0.999 + 0.001 * 0.99)
PaB

0.9098360655737705

Ответ: 0.9098360655737705